app/classifier/tasks/highlighter/index.jsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { Markdown } from 'markdownz';
+import GenericTaskEditor from '../generic-editor';
+import GenericTask from '../generic';
+import LabelRenderer from './label-renderer';
+import HighlighterSummary from './summary';
+import TextRenderer from '../../annotation-renderer/text';
+
+export default class Highlighter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.createButtons = this.createButtons.bind(this);
+    this.createLabelAnnotation = this.createLabelAnnotation.bind(this);
+  }
+
+  handleChange(toolIndex, e) {
+    let selection = document.getSelection();
+    this.createLabelAnnotation(selection, toolIndex);
+  }
+
+  createLabelAnnotation(selection, toolIndex) {
+    // currently we only deal with one selection at a time
+    const range = selection.getRangeAt(0);
+    const start = range.startOffset;
+    const end = range.endOffset;
+    const task = this.props.workflow.tasks[this.props.annotation.task];
+    const labelInformation = task.highlighterLabels[toolIndex];
+    const newAnnotation = Object.assign({}, this.props.annotation, { _toolIndex: toolIndex });
+    newAnnotation.value.push({
+      labelInformation: labelInformation,
+      start: start,
+      end: end
+    });
```

Comment:
The annotation should probably include the selected text as well as the start and end points.